lib/drivers/bluetoothLE/bluetoothLEDriver.js

```diff
@@ -383,13 +424,38 @@ module.exports = (config) => {
        return cb(err, null);
      }
 
-     progress(100);
-     return cb(null, data);
+     if (foraDriver) {
+       // Fora TN'G Voice uses a proprietary method to set device time
+       (async () => {
+         await foraDriver.init(cfg, data, cb);
+
+         cfg.abortTimer = setTimeout(() => {
+           debug('Time sync failed');
+           return cb(null, data); // not throwing error if time sync fails
+         }, 5000);
+
+         await foraDriver.syncDateTime(cfg);
+
+       })().catch((error) => {
+         debug('Error in time sync: ', error);
+         return cb(error, null);
+       });
+     } else {
+       progress(100);
+       return cb(null, data);
+     }
    },
 
    uploadData(progress, data, cb) {
      progress(0);
 
+     if (cfg.deviceInfo.name.startsWith('TNG')) {
+       // When Fora TN'G Voice meters are full (450 records), they add a bogus
+       // HI reading in 2037 to the end, so just use the first 450 records
```

Comment:
I'm so morbidly curious about what led to this being their solution to whatever problem they were trying to fix with this.

Reply:
I know, right?